src/rgw/rgw_rest_swift.cc

```diff
@@ -603,16 +603,15 @@ static int get_swift_versioning_settings(
     swift_ver_location = boost::in_place(std::string());
   }
 
-  std::string vloc = s->info.env->get("HTTP_X_VERSIONS_LOCATION", "");
-  if (vloc.size()) {
+  if (s->info.env->exists("HTTP_X_VERSIONS_LOCATION")) {
     /* If the Swift's versioning is globally disabled but someone wants to
      * enable it for a given container, new version of Swift will generate
      * the precondition failed error. */
     if (! s->cct->_conf->rgw_swift_versioning_enabled) {
       return -ERR_PRECONDITION_FAILED;
     }
 
-    swift_ver_location = std::move(vloc);
+    swift_ver_location = std::move(s->info.env->get("HTTP_X_VERSIONS_LOCATION", ""));
```

Comment:
Hmm, I would say that `std::move` isn't necessary anymore. The method already returns a temporary.